zeroex/order_validator_test.go

```diff
@@ -17,68 +18,150 @@ import (
+		ExpectedOrderStatus: InvalidTakerAssetData,
+	},
+	testCase{
+		Field:               "MakerAssetData",
+		Value:               malformedAssetData,
+		ExpectedOrderStatus: InvalidMakerAssetData,
+	},
+	testCase{
+		Field:               "TakerAssetData",
+		Value:               malformedAssetData,
+		ExpectedOrderStatus: InvalidTakerAssetData,
+	},
+	testCase{
+		Field:               "MakerAssetData",
+		Value:               unsupportedAssetData,
+		ExpectedOrderStatus: InvalidMakerAssetData,
+	},
+	testCase{
+		Field:               "TakerAssetData",
+		Value:               unsupportedAssetData,
+		ExpectedOrderStatus: InvalidTakerAssetData,
+	},
+	testCase{
+		Field:               "ExpirationTimeSeconds",
+		Value:               big.NewInt(time.Now().Add(-5 * time.Minute).Unix()),
+		ExpectedOrderStatus: Expired,
+	},
+}
+
+func TestBatchValidateOffChainCases(t *testing.T) {
 	rpcClient, err := rpc.Dial(constants.GanacheEndpoint)
 	require.NoError(t, err)
 
-	order := &Order{
-		MakerAddress:          constants.GanacheAccount0,
-		TakerAddress:          constants.NullAddress,
-		SenderAddress:         constants.NullAddress,
-		FeeRecipientAddress:   common.HexToAddress("0xa258b39954cef5cb142fd567a46cddb31a670124"),
-		MakerAssetData:        common.Hex2Bytes("f47261b000000000000000000000000034d402f14d58e001d8efbe6585051bf9706aa064"),
-		TakerAssetData:        common.Hex2Bytes("f47261b000000000000000000000000025b8fe1de9daf8ba351890744ff28cf7dfa8f5e3"),
-		Salt:                  big.NewInt(1548619145450),
-		MakerFee:              big.NewInt(0),
-		TakerFee:              big.NewInt(0),
-		MakerAssetAmount:      big.NewInt(3551808554499581700),
-		TakerAssetAmount:      big.NewInt(300000000000000),
-		ExpirationTimeSeconds: big.NewInt(1548619325),
-		ExchangeAddress:       contractNameToAddress.Exchange,
+	for _, testCase := range testCases {
+		order := modifyOrder(t, testOrder, testCase.Field, testCase.Value)
+
+		signedOrder, err := SignOrder(&order, rpcClient)
+		require.NoError(t, err)
+
+		orderHash, err := order.ComputeOrderHash()
+		require.NoError(t, err)
+
+		ethClient, err := ethclient.Dial(constants.GanacheEndpoint)
+		require.NoError(t, err)
+
+		signedOrders := []*SignedOrder{
+			signedOrder,
+		}
+
+		orderValidator, err := NewOrderValidator(ethClient, constants.TestNetworkID)
+		require.NoError(t, err)
+
+		orderInfos := orderValidator.BatchValidate(signedOrders)
+		assert.Len(t, orderInfos, 1)
+		assert.Equal(t, testCase.ExpectedOrderStatus, orderInfos[orderHash].OrderStatus)
+		assert.Equal(t, signedOrder, orderInfos[orderHash].SignedOrder)
 	}
-	signedOrder, err := SignOrder(order, rpcClient)
-	require.NoError(t, err)
+}
 
-	orderHash, err := order.ComputeOrderHash()
-	require.NoError(t, err)
+func TestBatchValidateMalformedSignature(t *testing.T) {
+	signedOrder := &SignedOrder{
+		Order: &testOrder,
+	}
+	// Incorrectly formatted signature
+	signedOrder.Signature = malformedSignature
```

Comment:
Is there a reason the malformed signature case can't be covered by a test case in `TestBatchValidateOffChainCases`?

Reply:
It was because all the test cases were for fields on `Orders`, and this changes the `SignedOrder`. I've refactored the test cases to use SignedOrders.